### PR TITLE
[master] Porting #49133 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #49133 to master

### CODE REVIEW

**Comment by garethgreenaway** [comment]
Needs changes from #55866 to allow PY3 tests to pass.

File: tests/unit/returners/test_postgres_local_cache.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            search_string = salt.utils.stringutils.to_str('return')
+            for call in connect_mock.mock_calls:
+                for arg in call.args:
+                    if isinstance(arg, tuple):
+                        for val in arg:
+                            if isinstance(val, six.string_types):
+                                if search_string in val:
+                                    return_val = json.loads(val)
+
+            self.assertNotEqual(return_val, None)
```
Comment (s0undt3ch) [inline tests/unit/returners/test_postgres_local_cache.py]: Please use `assertIsNotNone` instead

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by twangboy — DISMISSED** [review]

File: tests/unit/returners/test_postgres_local_cache.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        expected = {"return": "Trüe", "retcode": 0, "success": True}
+
+        connect_mock = MagicMock()
+        with patch.object(postgres_local_cache, "_get_conn", connect_mock):
+            postgres_local_cache.returner(load)
+
+            return_val = None
+            search_string = salt.utils.stringutils.to_str("return")
```
Comment (s0undt3ch) [inline tests/unit/returners/test_postgres_local_cache.py:115]:
```suggestion
            search_string = "return"
```
?

File: tests/unit/returners/test_postgres_local_cache.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+            self.assertIsNotNone(return_val, None)
+            self.assertDictEqual(return_val, expected)
+
+    def test_returner_unicode_exception(self):
+        """
+        Tests that the returner function
+        """
+        return_val = "Trüe"
+
```
Comment (s0undt3ch) [inline tests/unit/returners/test_postgres_local_cache.py:92]:
```suggestion
```

**Review by s0undt3ch — COMMENTED** [review]
Do you want to update to the comments I made?
The first change is an unused, and then overridden variable.
The second is an unnecessary call to `to_str`

**Review by garethgreenaway — COMMENTED** [review]
Committing requested changes.

**Review by s0undt3ch — DISMISSED** [review]

**Pull request merged**